Failed to build the bindings generator when cargo default to a cross-compiling toolchain

I'm working in a container that is used for cross compiling to aarch64-unknown-linux-gnu by including `build.target = "aarch64-unknown-linux-gnu"` inside of `${CARGO_HOME}/config`. This causes the bindings generator to fail to build. It should always specify the host target when invoking a secondary cargo command.

Operating system: Linux 5.7.8 in a podman container
OpenCV installation: manual cross-compilation
OpenCV version: 4.5.1
Rustc version: rustc 1.49.0 (e1884a8e3 2020-12-29)

Output of `cargo build -vv`
```
tmp> cargo new opencv-buildtime
     Created binary (application) `opencv-buildtime` package
tmp> cd opencv-buildtime
opencv-buildtime [master ?]> echo 'opencv = { version = "*", default-features = false, features = ["opencv-4", "buildtime-bindgen", "clang-runtime"] }' >> Cargo.toml
opencv-buildtime [master ?]> vi /rust/cargo/config
opencv-buildtime [master ?]> RUST_BACKTRACE=full cargo build -vv
    Updating crates.io index
       Fresh glob v0.3.0
       Fresh cfg-if v1.0.0
       Fresh lazy_static v1.4.0
       Fresh regex-syntax v0.6.22
       Fresh percent-encoding v2.1.0
       Fresh dunce v1.0.1
       Fresh maplit v1.0.2
       Fresh autocfg v1.0.1
       Fresh semver-parser v0.7.0
       Fresh pkg-config v0.3.19
       Fresh shlex v0.1.1
       Fresh vcpkg v0.2.11
   Compiling once_cell v1.5.2
   Compiling libc v0.2.82
       Fresh libloading v0.6.6
       Fresh thread_local v1.1.0
     Running `/target/debug/build/libc-dcaae1f179117036/build-script-build`
       Fresh semver v0.10.0
       Fresh memchr v2.3.4
   Compiling num-traits v0.2.14
     Running `CARGO=/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=once_cell CARGO_MANIFEST_DIR=/rust/cargo/registry/src/github.com-1ecc6299db9ec823/once_cell-1.5.2 CARGO_PKG_AUTHORS='Aleksey Kladov <aleksey.kladov@gmail.com>' CARGO_PKG_DESCRIPTION='Single assignment cells and lazy values.' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE='MIT OR Apache-2.0' CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=once_cell CARGO_PKG_REPOSITORY='https://github.com/matklad/once_cell' CARGO_PKG_VERSION=1.5.2 CARGO_PKG_VERSION_MAJOR=1 CARGO_PKG_VERSION_MINOR=5 CARGO_PKG_VERSION_PATCH=2 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/target/debug/deps:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib' /sccache/rustc rustc --crate-name once_cell --edition=2018 /rust/cargo/registry/src/github.com-1ecc6299db9ec823/once_cell-1.5.2/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="alloc"' --cfg 'feature="default"' --cfg 'feature="std"' -C metadata=d63e26e16d46d8d2 -C extra-filename=-d63e26e16d46d8d2 --out-dir /target/aarch64-unknown-linux-gnu/debug/deps --target aarch64-unknown-linux-gnu -C linker=rust-aarch64-linker-wrapper.sh -L dependency=/target/aarch64-unknown-linux-gnu/debug/deps -L dependency=/target/debug/deps --cap-lints warn`
       Fresh aho-corasick v0.7.15
   Compiling clang-sys v1.0.3
       Fresh jobserver v0.1.21
     Running `/target/debug/build/num-traits-db5b493d6092784a/build-script-build`
       Fresh regex v1.4.3
       Fresh cc v1.0.66
     Running `CARGO=/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=clang_sys CARGO_MANIFEST_DIR=/rust/cargo/registry/src/github.com-1ecc6299db9ec823/clang-sys-1.0.3 CARGO_PKG_AUTHORS='Kyle Mayes <kyle@mayeses.com>' CARGO_PKG_DESCRIPTION='Rust bindings for libclang.' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=Apache-2.0 CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=clang-sys CARGO_PKG_REPOSITORY='https://github.com/KyleMayes/clang-sys' CARGO_PKG_VERSION=1.0.3 CARGO_PKG_VERSION_MAJOR=1 CARGO_PKG_VERSION_MINOR=0 CARGO_PKG_VERSION_PATCH=3 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/target/debug/deps:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib' OUT_DIR=/target/debug/build/clang-sys-03dc036585bd8bb2/out /sccache/rustc rustc --crate-name clang_sys /rust/cargo/registry/src/github.com-1ecc6299db9ec823/clang-sys-1.0.3/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C codegen-units=256 --cfg 'feature="clang_3_5"' --cfg 'feature="clang_3_6"' --cfg 'feature="clang_3_7"' --cfg 'feature="clang_3_8"' --cfg 'feature="clang_3_9"' --cfg 'feature="clang_4_0"' --cfg 'feature="clang_5_0"' --cfg 'feature="clang_6_0"' --cfg 'feature="libloading"' --cfg 'feature="runtime"' -C metadata=aa7f159f2e12b8e1 -C extra-filename=-aa7f159f2e12b8e1 --out-dir /target/debug/deps -C linker=rust-x86_64-linker-wrapper.sh -L dependency=/target/debug/deps --extern glob=/target/debug/deps/libglob-1777aa6492c485b4.rmeta --extern libc=/target/debug/deps/liblibc-aecdbb85ac3dc19b.rmeta --extern libloading=/target/debug/deps/liblibloading-f82adc0b46a89c42.rmeta --cap-lints warn`
[libc 0.2.82] cargo:rustc-cfg=freebsd11
[libc 0.2.82] cargo:rustc-cfg=libc_priv_mod_use
[libc 0.2.82] cargo:rustc-cfg=libc_union
[libc 0.2.82] cargo:rustc-cfg=libc_const_size_of
[libc 0.2.82] cargo:rustc-cfg=libc_align
[libc 0.2.82] cargo:rustc-cfg=libc_core_cvoid
[libc 0.2.82] cargo:rustc-cfg=libc_packedN
[libc 0.2.82] cargo:rustc-cfg=libc_cfg_target_vendor
     Running `CARGO=/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=libc CARGO_MANIFEST_DIR=/rust/cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.82 CARGO_PKG_AUTHORS='The Rust Project Developers' CARGO_PKG_DESCRIPTION='Raw FFI bindings to platform libraries like libc.
' CARGO_PKG_HOMEPAGE='https://github.com/rust-lang/libc' CARGO_PKG_LICENSE='MIT OR Apache-2.0' CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=libc CARGO_PKG_REPOSITORY='https://github.com/rust-lang/libc' CARGO_PKG_VERSION=0.2.82 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=2 CARGO_PKG_VERSION_PATCH=82 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/target/debug/deps:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib' OUT_DIR=/target/aarch64-unknown-linux-gnu/debug/build/libc-34f43d58a733fbe5/out /sccache/rustc rustc --crate-name libc /rust/cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.82/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="default"' --cfg 'feature="std"' -C metadata=376cfb084fb76bfe -C extra-filename=-376cfb084fb76bfe --out-dir /target/aarch64-unknown-linux-gnu/debug/deps --target aarch64-unknown-linux-gnu -C linker=rust-aarch64-linker-wrapper.sh -L dependency=/target/aarch64-unknown-linux-gnu/debug/deps -L dependency=/target/debug/deps --cap-lints warn --cfg freebsd11 --cfg libc_priv_mod_use --cfg libc_union --cfg libc_const_size_of --cfg libc_align --cfg libc_core_cvoid --cfg libc_packedN --cfg libc_cfg_target_vendor`
[num-traits 0.2.14] cargo:rustc-cfg=has_i128
[num-traits 0.2.14] cargo:rustc-cfg=has_to_int_unchecked
[num-traits 0.2.14] cargo:rerun-if-changed=build.rs
     Running `CARGO=/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=num_traits CARGO_MANIFEST_DIR=/rust/cargo/registry/src/github.com-1ecc6299db9ec823/num-traits-0.2.14 CARGO_PKG_AUTHORS='The Rust Project Developers' CARGO_PKG_DESCRIPTION='Numeric traits for generic mathematics' CARGO_PKG_HOMEPAGE='https://github.com/rust-num/num-traits' CARGO_PKG_LICENSE='MIT OR Apache-2.0' CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=num-traits CARGO_PKG_REPOSITORY='https://github.com/rust-num/num-traits' CARGO_PKG_VERSION=0.2.14 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=2 CARGO_PKG_VERSION_PATCH=14 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/target/debug/deps:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib' OUT_DIR=/target/aarch64-unknown-linux-gnu/debug/build/num-traits-e0f6e07bc7f5cb17/out /sccache/rustc rustc --crate-name num_traits /rust/cargo/registry/src/github.com-1ecc6299db9ec823/num-traits-0.2.14/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="default"' --cfg 'feature="std"' -C metadata=2e10a761efa7b4c2 -C extra-filename=-2e10a761efa7b4c2 --out-dir /target/aarch64-unknown-linux-gnu/debug/deps --target aarch64-unknown-linux-gnu -C linker=rust-aarch64-linker-wrapper.sh -L dependency=/target/aarch64-unknown-linux-gnu/debug/deps -L dependency=/target/debug/deps --cap-lints warn --cfg has_i128 --cfg has_to_int_unchecked`
warning: associated function is never used: `discard`
   --> /target/debug/build/clang-sys-03dc036585bd8bb2/out/common.rs:104:12
    |
104 |     pub fn discard(mut self) {
    |            ^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

   Compiling clang v1.0.3
     Running `CARGO=/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=clang CARGO_MANIFEST_DIR=/rust/cargo/registry/src/github.com-1ecc6299db9ec823/clang-1.0.3 CARGO_PKG_AUTHORS='Kyle Mayes <kyle@mayeses.com>' CARGO_PKG_DESCRIPTION='A somewhat idiomatic Rust wrapper for libclang.' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=Apache-2.0 CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=clang CARGO_PKG_REPOSITORY='https://github.com/KyleMayes/clang-rs' CARGO_PKG_VERSION=1.0.3 CARGO_PKG_VERSION_MAJOR=1 CARGO_PKG_VERSION_MINOR=0 CARGO_PKG_VERSION_PATCH=3 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/target/debug/deps:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib' /sccache/rustc rustc --crate-name clang /rust/cargo/registry/src/github.com-1ecc6299db9ec823/clang-1.0.3/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C codegen-units=256 --cfg 'feature="clang_3_5"' --cfg 'feature="clang_3_6"' --cfg 'feature="clang_3_7"' --cfg 'feature="clang_3_8"' --cfg 'feature="clang_3_9"' --cfg 'feature="clang_4_0"' --cfg 'feature="clang_5_0"' --cfg 'feature="clang_6_0"' --cfg 'feature="runtime"' -C metadata=c41e612037d31883 -C extra-filename=-c41e612037d31883 --out-dir /target/debug/deps -C linker=rust-x86_64-linker-wrapper.sh -L dependency=/target/debug/deps --extern clang_sys=/target/debug/deps/libclang_sys-aa7f159f2e12b8e1.rmeta --extern libc=/target/debug/deps/liblibc-aecdbb85ac3dc19b.rmeta --cap-lints warn`
   Compiling opencv-binding-generator v0.22.0
     Running `CARGO=/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=opencv_binding_generator CARGO_MANIFEST_DIR=/rust/cargo/registry/src/github.com-1ecc6299db9ec823/opencv-binding-generator-0.22.0 CARGO_PKG_AUTHORS='Pro <twisted.fall@gmail.com>' CARGO_PKG_DESCRIPTION='Binding generator for opencv crate' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=opencv-binding-generator CARGO_PKG_REPOSITORY='https://github.com/twistedfall/opencv-rust' CARGO_PKG_VERSION=0.22.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=22 CARGO_PKG_VERSION_PATCH=0 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/target/debug/deps:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib' /sccache/rustc rustc --crate-name opencv_binding_generator --edition=2018 /rust/cargo/registry/src/github.com-1ecc6299db9ec823/opencv-binding-generator-0.22.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C codegen-units=256 -C metadata=cb734a570223c72a -C extra-filename=-cb734a570223c72a --out-dir /target/debug/deps -C linker=rust-x86_64-linker-wrapper.sh -L dependency=/target/debug/deps --extern clang=/target/debug/deps/libclang-c41e612037d31883.rmeta --extern clang_sys=/target/debug/deps/libclang_sys-aa7f159f2e12b8e1.rmeta --extern dunce=/target/debug/deps/libdunce-367da047d5f20893.rmeta --extern maplit=/target/debug/deps/libmaplit-b336eb0bd13ad66a.rmeta --extern once_cell=/target/debug/deps/libonce_cell-2c92c24fa9602637.rmeta --extern percent_encoding=/target/debug/deps/libpercent_encoding-00fa30bf8463aff7.rmeta --extern regex=/target/debug/deps/libregex-732d9979a57587ee.rmeta --cap-lints warn`
warning: 1 warning emitted

   Compiling opencv v0.47.0
     Running `CARGO=/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=build_script_build CARGO_MANIFEST_DIR=/rust/cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.47.0 CARGO_PKG_AUTHORS='Mathieu Poumeyrol <kali@zoy.org>:Pro <twisted.fall@gmail.com>' CARGO_PKG_DESCRIPTION='Rust bindings for OpenCV' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=opencv CARGO_PKG_REPOSITORY='https://github.com/twistedfall/opencv-rust' CARGO_PKG_VERSION=0.47.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=47 CARGO_PKG_VERSION_PATCH=0 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/target/debug/deps:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib:/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib' /sccache/rustc rustc --crate-name build_script_build --edition=2018 /rust/cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.47.0/build.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type bin --emit=dep-info,link -C opt-level=3 -C embed-bitcode=no -C codegen-units=256 --cfg 'feature="binding-generator"' --cfg 'feature="buildtime-bindgen"' --cfg 'feature="clang"' --cfg 'feature="clang-runtime"' --cfg 'feature="jobserver"' --cfg 'feature="opencv-4"' -C metadata=23c98089364e7e11 -C extra-filename=-23c98089364e7e11 --out-dir /target/debug/build/opencv-23c98089364e7e11 -C linker=rust-x86_64-linker-wrapper.sh -L dependency=/target/debug/deps --extern cc=/target/debug/deps/libcc-e316c70b6ec8942e.rlib --extern clang=/target/debug/deps/libclang-c41e612037d31883.rlib --extern dunce=/target/debug/deps/libdunce-367da047d5f20893.rlib --extern glob=/target/debug/deps/libglob-1777aa6492c485b4.rlib --extern jobserver=/target/debug/deps/libjobserver-141749752d9795c0.rlib --extern once_cell=/target/debug/deps/libonce_cell-2c92c24fa9602637.rlib --extern binding_generator=/target/debug/deps/libopencv_binding_generator-cb734a570223c72a.rlib --extern pkg_config=/target/debug/deps/libpkg_config-23614dd08f52df50.rlib --extern semver=/target/debug/deps/libsemver-a86ecbdf4b6df1f8.rlib --extern shlex=/target/debug/deps/libshlex-7244f3fbe35e4cd0.rlib --extern vcpkg=/target/debug/deps/libvcpkg-5d805d9847331388.rlib --cap-lints warn`
     Running `/target/debug/build/opencv-23c98089364e7e11/build-script-build`
[opencv 0.47.0] running: "/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/bin/cargo" "build" "--release" "--package" "opencv-binding-generator" "--bin" "binding-generator"
[opencv 0.47.0] === Crate version: Some("0.47.0")
[opencv 0.47.0] === Environment configuration:
[opencv 0.47.0] ===   OPENCV_HEADER_DIR = None
[opencv 0.47.0] ===   OPENCV_PACKAGE_NAME = None
[opencv 0.47.0] ===   OPENCV_PKGCONFIG_NAME = None
[opencv 0.47.0] ===   OPENCV_CMAKE_NAME = None
[opencv 0.47.0] ===   OPENCV_CMAKE_BIN = None
[opencv 0.47.0] ===   OPENCV_VCPKG_NAME = None
[opencv 0.47.0] ===   OPENCV_LINK_LIBS = None
[opencv 0.47.0] ===   OPENCV_LINK_PATHS = None
[opencv 0.47.0] ===   OPENCV_INCLUDE_PATHS = None
[opencv 0.47.0] ===   OPENCV_DISABLE_PROBES = None
[opencv 0.47.0] ===   OPENCV_CLANG_STDLIB_PATH = None
[opencv 0.47.0] ===   OPENCV_MODULE_WHITELIST = None
[opencv 0.47.0] ===   OPENCV_MODULE_BLACKLIST = None
[opencv 0.47.0] ===   CMAKE_PREFIX_PATH = None
[opencv 0.47.0] ===   OpenCV_DIR = None
[opencv 0.47.0] ===   PKG_CONFIG_PATH = None
[opencv 0.47.0] ===   VCPKG_ROOT = None
[opencv 0.47.0] ===   VCPKGRS_DYNAMIC = None
[opencv 0.47.0] === Enabled features:
[opencv 0.47.0] ===   BINDING_GENERATOR
[opencv 0.47.0] ===   BUILDTIME_BINDGEN
[opencv 0.47.0] ===   CLANG
[opencv 0.47.0] ===   CLANG_RUNTIME
[opencv 0.47.0] ===   JOBSERVER
[opencv 0.47.0] ===   OPENCV_4
[opencv 0.47.0] === Detected probe priority based on environment vars: pkg_config: false, cmake: false, vcpkg: false
[opencv 0.47.0] === Probing the OpenCV library in the following order: environment, pkg_config, cmake, vcpkg_cmake, vcpkg
[opencv 0.47.0] === Can't probe using: environment, continuing with other methods because: Some environment variables are missing
[opencv 0.47.0] === Probing OpenCV library using pkg_config
[opencv 0.47.0] cargo:rerun-if-env-changed=OPENCV4_NO_PKG_CONFIG
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_aarch64-unknown-linux-gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_aarch64_unknown_linux_gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG
[opencv 0.47.0] cargo:rerun-if-env-changed=OPENCV4_STATIC
[opencv 0.47.0] cargo:rerun-if-env-changed=OPENCV4_DYNAMIC
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64-unknown-linux-gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64_unknown_linux_gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64-unknown-linux-gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64_unknown_linux_gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_LIBDIR
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64-unknown-linux-gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64_unknown_linux_gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
[opencv 0.47.0] cargo:rerun-if-env-changed=SYSROOT
[opencv 0.47.0] cargo:rerun-if-env-changed=OPENCV4_STATIC
[opencv 0.47.0] cargo:rerun-if-env-changed=OPENCV4_DYNAMIC
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG
[opencv 0.47.0] cargo:rerun-if-env-changed=OPENCV4_STATIC
[opencv 0.47.0] cargo:rerun-if-env-changed=OPENCV4_DYNAMIC
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64-unknown-linux-gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64_unknown_linux_gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64-unknown-linux-gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64_unknown_linux_gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_LIBDIR
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64-unknown-linux-gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64_unknown_linux_gnu
[opencv 0.47.0] cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
[opencv 0.47.0] cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
[opencv 0.47.0] === OpenCV library configuration: Library {
[opencv 0.47.0]     include_paths: [
[opencv 0.47.0]         "/aarch64/opencv/include/opencv4",
[opencv 0.47.0]     ],
[opencv 0.47.0]     version: "4.5.1",
[opencv 0.47.0]     cargo_metadata: [
[opencv 0.47.0]         "cargo:rustc-link-search=/aarch64/opencv/lib",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_gapi",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_highgui",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_ml",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_objdetect",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_photo",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_stitching",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_video",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_calib3d",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_features2d",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_dnn",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_flann",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_videoio",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_imgcodecs",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_imgproc",
[opencv 0.47.0]         "cargo:rustc-link-lib=opencv_core",
[opencv 0.47.0]     ],
[opencv 0.47.0] }
[opencv 0.47.0] === Found OpenCV library version: 4.5.1 in headers located at: /aarch64/opencv/include/opencv4
[opencv 0.47.0] === Using OpenCV headers from: /aarch64/opencv/include/opencv4/opencv2
[opencv 0.47.0] === Generating code in: /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out
[opencv 0.47.0] === Placing generated bindings into: /rust/cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.47.0/bindings/rust/opencv_4
[opencv 0.47.0] === Clang: clang version 11.0.0 (Fedora 11.0.0-1.fc33)
[opencv 0.47.0] === Clang command line args: [
[opencv 0.47.0]     "-isystem/usr/bin/../lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10",
[opencv 0.47.0]     "-isystem/usr/bin/../lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/x86_64-redhat-linux",
[opencv 0.47.0]     "-isystem/usr/bin/../lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/backward",
[opencv 0.47.0]     "-isystem/usr/local/include",
[opencv 0.47.0]     "-isystem/usr/lib64/clang/11.0.0/include",
[opencv 0.47.0]     "-isystem/usr/include",
[opencv 0.47.0]     "-I/aarch64/opencv/include/opencv4",
[opencv 0.47.0]     "-I/rust/cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.47.0/src_cpp",
[opencv 0.47.0]     "-DOCVRS_PARSING_HEADERS",
[opencv 0.47.0]     "-includeocvrs_resolve_types.hpp",
[opencv 0.47.0]     "-std=c++14",
[opencv 0.47.0] ]
[opencv 0.47.0]    Compiling glob v0.3.0
[opencv 0.47.0]    Compiling memchr v2.3.4
[opencv 0.47.0]    Compiling libc v0.2.80
[opencv 0.47.0]    Compiling lazy_static v1.4.0
[opencv 0.47.0]    Compiling regex-syntax v0.6.21
[opencv 0.47.0]    Compiling dunce v1.0.1
[opencv 0.47.0]    Compiling maplit v1.0.2
[opencv 0.47.0]    Compiling once_cell v1.5.2
[opencv 0.47.0]    Compiling percent-encoding v2.1.0
[opencv 0.47.0]    Compiling thread_local v1.0.1
[opencv 0.47.0]    Compiling aho-corasick v0.7.15
[opencv 0.47.0]    Compiling clang-sys v1.0.3
[opencv 0.47.0]    Compiling regex v1.4.2
[opencv 0.47.0]    Compiling clang v1.0.2
[opencv 0.47.0]    Compiling opencv-binding-generator v0.22.0
[opencv 0.47.0] error: linking with `rust-aarch64-linker-wrapper.sh` failed: exit code: 1
[opencv 0.47.0]   |
[opencv 0.47.0]   = note: "rust-aarch64-linker-wrapper.sh" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-Wl,--eh-frame-hdr" "-L" "/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps/binding_generator-ffe1adac119350ef.binding_generator.6nu1qu8h-cgu.2.rcgu.o" "-o" "/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps/binding_generator-ffe1adac119350ef" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-Wl,-O1" "-nodefaultlibs" "-L" "/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps" "-L" "/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/release/deps" "-L" "/usr/lib64" "-L" "/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-Wl,--start-group" "-Wl,--end-group" "-Wl,-Bstatic" "/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcompiler_builtins-dedf96718b1b5f55.rlib" "-Wl,-Bdynamic" "-lclang" "-lutil" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-lutil"
[opencv 0.47.0]   = note: /usr/local/aarch64-linux-gnu/bin/../lib/gcc/aarch64-none-linux-gnu/10.2.1/../../../../aarch64-none-linux-gnu/bin/ld.gold: fatal error: /usr/lib64/libclang.so: unsupported ELF machine number 62
[opencv 0.47.0]           collect2: error: ld returned 1 exit status
[opencv 0.47.0]           running /sccache/aarch64-linux-gnu-gcc -fuse-ld=gold -Wl,--as-needed -Wl,-z,noexecstack -Wl,--eh-frame-hdr -L /rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps/binding_generator-ffe1adac119350ef.binding_generator.6nu1qu8h-cgu.2.rcgu.o -o /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps/binding_generator-ffe1adac119350ef -Wl,--gc-sections -pie -Wl,-zrelro -Wl,-znow -Wl,-O1 -nodefaultlibs -L /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps -L /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/release/deps -L /usr/lib64 -L /rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib -Wl,--start-group -Wl,--end-group -Wl,-Bstatic /rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcompiler_builtins-dedf96718b1b5f55.rlib -Wl,-Bdynamic -lclang -lutil -lgcc_s -lutil -lrt -lpthread -lm -ldl -lc -lutil -Wl,-rpath-link,/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib -Wl,-rpath-link,/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps -Wl,-rpath-link,/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/release/deps -Wl,-rpath-link,/usr/lib64 -Wl,-rpath-link,/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib
[opencv 0.47.0]
[opencv 0.47.0]
[opencv 0.47.0] error: aborting due to previous error
[opencv 0.47.0]
[opencv 0.47.0] error: could not compile `opencv-binding-generator`
[opencv 0.47.0]
[opencv 0.47.0] To learn more, run the command again with --verbose.
[opencv 0.47.0] Error: "Failed to build the bindings generator"
error: failed to run custom build command for `opencv v0.47.0`

Caused by:
  process didn't exit successfully: `/target/debug/build/opencv-23c98089364e7e11/build-script-build` (exit code: 1)
  --- stdout
  running: "/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/bin/cargo" "build" "--release" "--package" "opencv-binding-generator" "--bin" "binding-generator"
  cargo:rerun-if-env-changed=OPENCV4_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_aarch64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS_aarch64_unknown_linux_gnu
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG_ALLOW_CROSS
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=OPENCV4_STATIC
  cargo:rerun-if-env-changed=OPENCV4_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64_unknown_linux_gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64_unknown_linux_gnu
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64_unknown_linux_gnu
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=SYSROOT
  cargo:rerun-if-env-changed=OPENCV4_STATIC
  cargo:rerun-if-env-changed=OPENCV4_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=OPENCV4_STATIC
  cargo:rerun-if-env-changed=OPENCV4_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_aarch64_unknown_linux_gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_aarch64_unknown_linux_gnu
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_aarch64_unknown_linux_gnu
  cargo:rerun-if-env-changed=TARGET_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  === Clang: clang version 11.0.0 (Fedora 11.0.0-1.fc33)

  --- stderr
  === Crate version: Some("0.47.0")
  === Environment configuration:
  ===   OPENCV_HEADER_DIR = None
  ===   OPENCV_PACKAGE_NAME = None
  ===   OPENCV_PKGCONFIG_NAME = None
  ===   OPENCV_CMAKE_NAME = None
  ===   OPENCV_CMAKE_BIN = None
  ===   OPENCV_VCPKG_NAME = None
  ===   OPENCV_LINK_LIBS = None
  ===   OPENCV_LINK_PATHS = None
  ===   OPENCV_INCLUDE_PATHS = None
  ===   OPENCV_DISABLE_PROBES = None
  ===   OPENCV_CLANG_STDLIB_PATH = None
  ===   OPENCV_MODULE_WHITELIST = None
  ===   OPENCV_MODULE_BLACKLIST = None
  ===   CMAKE_PREFIX_PATH = None
  ===   OpenCV_DIR = None
  ===   PKG_CONFIG_PATH = None
  ===   VCPKG_ROOT = None
  ===   VCPKGRS_DYNAMIC = None
  === Enabled features:
  ===   BINDING_GENERATOR
  ===   BUILDTIME_BINDGEN
  ===   CLANG
  ===   CLANG_RUNTIME
  ===   JOBSERVER
  ===   OPENCV_4
  === Detected probe priority based on environment vars: pkg_config: false, cmake: false, vcpkg: false
  === Probing the OpenCV library in the following order: environment, pkg_config, cmake, vcpkg_cmake, vcpkg
  === Can't probe using: environment, continuing with other methods because: Some environment variables are missing
  === Probing OpenCV library using pkg_config
  === OpenCV library configuration: Library {
      include_paths: [
          "/aarch64/opencv/include/opencv4",
      ],
      version: "4.5.1",
      cargo_metadata: [
          "cargo:rustc-link-search=/aarch64/opencv/lib",
          "cargo:rustc-link-lib=opencv_gapi",
          "cargo:rustc-link-lib=opencv_highgui",
          "cargo:rustc-link-lib=opencv_ml",
          "cargo:rustc-link-lib=opencv_objdetect",
          "cargo:rustc-link-lib=opencv_photo",
          "cargo:rustc-link-lib=opencv_stitching",
          "cargo:rustc-link-lib=opencv_video",
          "cargo:rustc-link-lib=opencv_calib3d",
          "cargo:rustc-link-lib=opencv_features2d",
          "cargo:rustc-link-lib=opencv_dnn",
          "cargo:rustc-link-lib=opencv_flann",
          "cargo:rustc-link-lib=opencv_videoio",
          "cargo:rustc-link-lib=opencv_imgcodecs",
          "cargo:rustc-link-lib=opencv_imgproc",
          "cargo:rustc-link-lib=opencv_core",
      ],
  }
  === Found OpenCV library version: 4.5.1 in headers located at: /aarch64/opencv/include/opencv4
  === Using OpenCV headers from: /aarch64/opencv/include/opencv4/opencv2
  === Generating code in: /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out
  === Placing generated bindings into: /rust/cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.47.0/bindings/rust/opencv_4
  === Clang command line args: [
      "-isystem/usr/bin/../lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10",
      "-isystem/usr/bin/../lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/x86_64-redhat-linux",
      "-isystem/usr/bin/../lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/backward",
      "-isystem/usr/local/include",
      "-isystem/usr/lib64/clang/11.0.0/include",
      "-isystem/usr/include",
      "-I/aarch64/opencv/include/opencv4",
      "-I/rust/cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.47.0/src_cpp",
      "-DOCVRS_PARSING_HEADERS",
      "-includeocvrs_resolve_types.hpp",
      "-std=c++14",
  ]
     Compiling glob v0.3.0
     Compiling memchr v2.3.4
     Compiling libc v0.2.80
     Compiling lazy_static v1.4.0
     Compiling regex-syntax v0.6.21
     Compiling dunce v1.0.1
     Compiling maplit v1.0.2
     Compiling once_cell v1.5.2
     Compiling percent-encoding v2.1.0
     Compiling thread_local v1.0.1
     Compiling aho-corasick v0.7.15
     Compiling clang-sys v1.0.3
     Compiling regex v1.4.2
     Compiling clang v1.0.2
     Compiling opencv-binding-generator v0.22.0
  error: linking with `rust-aarch64-linker-wrapper.sh` failed: exit code: 1
    |
    = note: "rust-aarch64-linker-wrapper.sh" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-Wl,--eh-frame-hdr" "-L" "/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps/binding_generator-ffe1adac119350ef.binding_generator.6nu1qu8h-cgu.2.rcgu.o" "-o" "/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps/binding_generator-ffe1adac119350ef" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-Wl,-O1" "-nodefaultlibs" "-L" "/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps" "-L" "/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/release/deps" "-L" "/usr/lib64" "-L" "/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-Wl,--start-group" "-Wl,--end-group" "-Wl,-Bstatic" "/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcompiler_builtins-dedf96718b1b5f55.rlib" "-Wl,-Bdynamic" "-lclang" "-lutil" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-lutil"
    = note: /usr/local/aarch64-linux-gnu/bin/../lib/gcc/aarch64-none-linux-gnu/10.2.1/../../../../aarch64-none-linux-gnu/bin/ld.gold: fatal error: /usr/lib64/libclang.so: unsupported ELF machine number 62
            collect2: error: ld returned 1 exit status
            running /sccache/aarch64-linux-gnu-gcc -fuse-ld=gold -Wl,--as-needed -Wl,-z,noexecstack -Wl,--eh-frame-hdr -L /rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps/binding_generator-ffe1adac119350ef.binding_generator.6nu1qu8h-cgu.2.rcgu.o -o /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps/binding_generator-ffe1adac119350ef -Wl,--gc-sections -pie -Wl,-zrelro -Wl,-znow -Wl,-O1 -nodefaultlibs -L /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps -L /target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/release/deps -L /usr/lib64 -L /rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib -Wl,--start-group -Wl,--end-group -Wl,-Bstatic /rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcompiler_builtins-dedf96718b1b5f55.rlib -Wl,-Bdynamic -lclang -lutil -lgcc_s -lutil -lrt -lpthread -lm -ldl -lc -lutil -Wl,-rpath-link,/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib -Wl,-rpath-link,/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/aarch64-unknown-linux-gnu/release/deps -Wl,-rpath-link,/target/aarch64-unknown-linux-gnu/debug/build/opencv-e66a88d44115a898/out/release/deps -Wl,-rpath-link,/usr/lib64 -Wl,-rpath-link,/rust/rustup/toolchains/1.49.0-x86_64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib


  error: aborting due to previous error

  error: could not compile `opencv-binding-generator`

  To learn more, run the command again with --verbose.
  Error: "Failed to build the bindings generator"
```

--- a/build.rs
+++ b/build.rs
@@ -59,7 +59,7 @@ static OUT_DIR: Lazy<PathBuf> = Lazy::new(|| PathBuf::from(env::var_os("OUT_DIR"
 static MANIFEST_DIR: Lazy<PathBuf> = Lazy::new(|| PathBuf::from(env::var_os("CARGO_MANIFEST_DIR").expect("Can't read CARGO_MANIFEST_DIR env var")));
 static SRC_DIR: Lazy<PathBuf> = Lazy::new(|| MANIFEST_DIR.join("src"));
 static SRC_CPP_DIR: Lazy<PathBuf> = Lazy::new(|| MANIFEST_DIR.join("src_cpp"));
-static HOST_TRIPLE: Lazy<String> = Lazy::new(||env::var("HOST_TRIPLE").expect("Unable to determine host triple"));
+static HOST_TRIPLE: Lazy<Option<String>> = Lazy::new(||env::var("HOST_TRIPLE").ok());
 
 static ENV_VARS: [&str; 18] = [
 	"OPENCV_HEADER_DIR",
@@ -818,8 +818,11 @@ fn main() -> Result<()> {
 		let cargo_bin = PathBuf::from(env::var_os("CARGO").unwrap_or_else(|| "cargo".into()));
 		let mut cargo = Command::new(cargo_bin);
 		// generator script is quite slow in debug mode, so we force it to be built in release mode
-		cargo.args(&["build", "--release", "--package", "opencv-binding-generator", "--bin", "binding-generator", "--target", &*HOST_TRIPLE])
+		cargo.args(&["build", "--release", "--package", "opencv-binding-generator", "--bin", "binding-generator"])
 			.env("CARGO_TARGET_DIR", &*OUT_DIR);
+		if let Some(host_triple) = HOST_TRIPLE.as_ref() {
+			cargo.args(&["--target", host_triple]);
+		}
 		println!("running: {:?}", &cargo);
 		Some(cargo.spawn()?)
 	} else {

--- a/build.rs
+++ b/build.rs
@@ -59,6 +59,7 @@ static OUT_DIR: Lazy<PathBuf> = Lazy::new(|| PathBuf::from(env::var_os("OUT_DIR"
 static MANIFEST_DIR: Lazy<PathBuf> = Lazy::new(|| PathBuf::from(env::var_os("CARGO_MANIFEST_DIR").expect("Can't read CARGO_MANIFEST_DIR env var")));
 static SRC_DIR: Lazy<PathBuf> = Lazy::new(|| MANIFEST_DIR.join("src"));
 static SRC_CPP_DIR: Lazy<PathBuf> = Lazy::new(|| MANIFEST_DIR.join("src_cpp"));
+static HOST_TRIPLE: Lazy<String> = Lazy::new(||env::var("HOST_TRIPLE").expect("Unable to determine host triple"));
 
 static ENV_VARS: [&str; 18] = [
 	"OPENCV_HEADER_DIR",
@@ -817,7 +818,7 @@ fn main() -> Result<()> {
 		let cargo_bin = PathBuf::from(env::var_os("CARGO").unwrap_or_else(|| "cargo".into()));
 		let mut cargo = Command::new(cargo_bin);
 		// generator script is quite slow in debug mode, so we force it to be built in release mode
-		cargo.args(&["build", "--release", "--package", "opencv-binding-generator", "--bin", "binding-generator"])
+		cargo.args(&["build", "--release", "--package", "opencv-binding-generator", "--bin", "binding-generator", "--target", &*HOST_TRIPLE])
 			.env("CARGO_TARGET_DIR", &*OUT_DIR);
 		println!("running: {:?}", &cargo);
 		Some(cargo.spawn()?)

--- a/build_generator.rs
+++ b/build_generator.rs
@@ -16,6 +16,7 @@ use super::{
 	file_copy_to_dir,
 	get_versioned_hub_dirs,
 	is_core_module,
+	HOST_TRIPLE,
 	MODULES,
 	OUT_DIR,
 	Result,
@@ -110,7 +111,7 @@ pub fn gen_wrapper(opencv_header_dir: &Path, generator_build: Option<Child>) -> 
 					let clang_stdlib_include_dir = (*clang_stdlib_include_dir).as_ref()
 						.and_then(|p| p.to_str())
 						.unwrap_or("None");
-					let mut bin_generator = Command::new(OUT_DIR.join("release/binding-generator"));
+					let mut bin_generator = Command::new(OUT_DIR.join(format!("{}/release/binding-generator", &*HOST_TRIPLE)));
 					bin_generator.arg(&*opencv_header_dir)
 						.arg(&*SRC_CPP_DIR)
 						.arg(&*OUT_DIR)

--- a/build_generator.rs
+++ b/build_generator.rs
@@ -111,7 +111,10 @@ pub fn gen_wrapper(opencv_header_dir: &Path, generator_build: Option<Child>) -> 
 					let clang_stdlib_include_dir = (*clang_stdlib_include_dir).as_ref()
 						.and_then(|p| p.to_str())
 						.unwrap_or("None");
-					let mut bin_generator = Command::new(OUT_DIR.join(format!("{}/release/binding-generator", &*HOST_TRIPLE)));
+					let mut bin_generator = match HOST_TRIPLE.as_ref() {
+						Some(host_triple) => Command::new(OUT_DIR.join(format!("{}/release/binding-generator", host_triple))),
+						None => Command::new(OUT_DIR.join("release/binding-generator")),
+					};
 					bin_generator.arg(&*opencv_header_dir)
 						.arg(&*SRC_CPP_DIR)
 						.arg(&*OUT_DIR)

